pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -2342,15 +2342,19 @@ protected SchemaCompatibilityStrategy internalGetSchemaCompatibilityStrategy() {
         validateNamespacePolicyOperation(namespaceName, PolicyName.SCHEMA_COMPATIBILITY_STRATEGY,
                 PolicyOperation.READ);
         Policies policies = getNamespacePolicies(namespaceName);
+
         SchemaCompatibilityStrategy schemaCompatibilityStrategy = policies.schema_compatibility_strategy;
-        if (schemaCompatibilityStrategy == SchemaCompatibilityStrategy.UNDEFINED) {
-            schemaCompatibilityStrategy = pulsar().getConfig().getSchemaCompatibilityStrategy();
-            if (schemaCompatibilityStrategy == SchemaCompatibilityStrategy.UNDEFINED) {
-                schemaCompatibilityStrategy = SchemaCompatibilityStrategy
-                        .fromAutoUpdatePolicy(policies.schema_auto_update_compatibility_strategy);
-            }
+        if (!SchemaCompatibilityStrategy.isUndefined(schemaCompatibilityStrategy)) {
+            return schemaCompatibilityStrategy;
         }
-        return schemaCompatibilityStrategy;
+
+        schemaCompatibilityStrategy =
+                SchemaCompatibilityStrategy.fromAutoUpdatePolicy(policies.schema_auto_update_compatibility_strategy);
+        if (!SchemaCompatibilityStrategy.isUndefined(schemaCompatibilityStrategy)) {
+            return schemaCompatibilityStrategy;
+        }
+
+        return pulsar().getConfig().getSchemaCompatibilityStrategy();
```

Comment:
For getting the namespace policy, we should not return the broker-level configuration.